runtime/compiler/env/J9SharedCache.cpp

```diff
@@ -1567,4 +1571,108 @@ TR_J9JITServerSharedCache::storeSharedData(J9VMThread *vmThread, const char *key
    return std::get<0>(_stream->read<const void *>());
    }
 
+TR_J9DeserializerSharedCache::TR_J9DeserializerSharedCache(TR_J9VMBase *fe, JITServerNoSCCAOTDeserializer *deserializer)
+   : TR_J9SharedCache(fe)
+   {
+   _deserializer = deserializer;
+   }
+
+J9ROMClass *
+TR_J9DeserializerSharedCache::romClassFromOffsetInSharedCache(uintptr_t offset)
+   {
+   TR::Compilation *comp = TR::compInfoPT->getCompilation();
+   bool wasReset = false;
+   auto romClass = _deserializer->romClassFromOffsetInSharedCache(offset, comp, wasReset);
+   if (wasReset)
+      comp->failCompilation<J9::AOTDeserializerReset>(
+         "Deserializer reset during relocation of method %s", comp->signature());
+   TR_ASSERT_FATAL(romClass, "ROM class for offset %zu could not be found",
+                   offset,
+                   JITServerNoSCCAOTDeserializer::offsetId(offset),
+                   JITServerNoSCCAOTDeserializer::offsetType(offset));
+   return romClass;
+   }
+
+void *
+TR_J9DeserializerSharedCache::pointerFromOffsetInSharedCache(uintptr_t offset)
+   {
+   TR::Compilation *comp = TR::compInfoPT->getCompilation();
+   bool wasReset = false;
+   auto ptr = _deserializer->pointerFromOffsetInSharedCache(offset, comp, wasReset);
+   if (wasReset)
+      comp->failCompilation<J9::AOTDeserializerReset>(
+         "Deserializer reset during relocation of method %s", comp->signature());
+   TR_ASSERT_FATAL(ptr, "Pointer for offset %zu ID %zu type %u could not be found",
+                   offset,
+                   JITServerNoSCCAOTDeserializer::offsetId(offset),
+                   JITServerNoSCCAOTDeserializer::offsetType(offset));
+   return ptr;
+   }
+
+void *
+TR_J9DeserializerSharedCache::lookupClassLoaderAssociatedWithClassChain(void *chainData)
```

Comment:
This method really does just return `chainData`, because the only thing it's called on is the result of `pointerFromOffsetInSharedCache` when applied to a class loader offset. Maybe it would be better to add a dedicated `TR_J9SharedCache::lookupClassLoaderFromLoaderOffset()` that would get a `J9ClassLoader *` directly from a class loader offset? It would avoid this sort of `void *` trickery.